onyo/lib/onyo.py

```diff
@@ -510,6 +512,32 @@ def _fsck_clean_tree(self) -> bool:
 
         return True
 
+    def _fsck_no_pseudo_keys(self) -> bool:
+        """
+        Check that no asset contains any pseudo-key names.
+        """
+        assets_with_pseudo_keys = {}
+        pseudo_keys = ["type", "make", "model", "serial"]
+
+        for asset in self.assets:
+            contents = self._read_asset(asset)
+
+            for p in pseudo_keys:
+                if p in contents:
+                    if asset not in assets_with_pseudo_keys.keys():
+                        assets_with_pseudo_keys[asset] = []
+                    assets_with_pseudo_keys[asset].append(p)
+
+        if assets_with_pseudo_keys:
+            log.error('The following assets contain pseudo keys:\n' +
+                      self._n_join([f'{asset}: ' +
+                                    ', '.join([str(k) for k in assets_with_pseudo_keys[asset]])
+                                    for asset in list(assets_with_pseudo_keys.keys())]))
```

Comment:
`.keys()` is not necessary to loop over a dict. And if it were, `.keys()` would not need to be cast into a list in order to iterate over it.

Needless casting is a common error across these PRs. Please only do so when you know that it is necessary.